tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -47,9 +47,10 @@ class RNTupleColumnReader final : public ROOT::Detail::RDF::RColumnReaderBase {
       const auto &descriptor = source.GetDescriptor();
       const auto fieldId = descriptor.FindFieldId(colName);
       const auto &fieldDescriptor = descriptor.GetFieldDescriptor(fieldId);
-      const auto typeName = fieldDescriptor.GetTypeName();
-      auto fieldBasePtr = Detail::RFieldBase::Create(fieldDescriptor.GetFieldName(), typeName).Unwrap();
-      Detail::RFieldFuse::ConnectRecursively(fieldId, source, *fieldBasePtr);
+      auto fieldBasePtr = fieldDescriptor.CreateField(descriptor);
+      fieldBasePtr->ConnectPageStorage(source);
+      for (auto &f : *fieldBasePtr)
```

Comment:
This seems to imply to ConnectPageStorage is not recursive but the loop below goes only one level 'deep'.  Is there only one level of nesting to the fields?

Reply:
That's right, `ConnectPageStorage` is not recursive but the field iterator traverses the entire subtree so the net effect is the same than the previous `Detail::RFieldFuse::ConnectRecursively`.